figtree.go

```diff
@@ -29,26 +29,22 @@ type Logger interface {
 var Log Logger = logging.MustGetLogger("figtree")
 
 type FigTree struct {
+	Home      string
+	WorkDir   string
 	ConfigDir string
 	Defaults  interface{}
 	EnvPrefix string
 	stop      bool
 }
 
-func NewFigTree() *FigTree {
+func NewFigTree(home, workDir string) *FigTree {
```

Comment:
I am wondering if we can/should leave the NewFigTree fuction prototype alone and create another `New` function, something like:
```
func NewFigTree() *FigTree {
  wd, _  := os.Getwd()
  return &FigTree{
    Home: os.Getenv("HOME"),
    WorkDir: wd,
    EnvPrefix: "FIGTREE",
  }
}

func NewFigTreeForPath(home, workDir string) * FigTree {
  return &FigTree{Home: home, WorkDir: workDir, EnvPrefix: "FIGTREE"}
}
```

Not sure about the `NewFigTreeForPath` name, open to suggestions there.  We could also just skip the `NewFigTreeForPath` function completely since it is jut a one-liner wrapper for a struct creation with public variables.  Users could just create the struct manaully as desired without this helper function.

Then we could add back in the figtree.LoadAllConfigs and figtree.LoadConfig helper functions.

Reply:
The main reason why I wanted to remove the `figtree.LoadAllConfigs` and `figtree.LoadConfig` as well as breaking the function prototype was to ensure that all calls will have compile errors without explicitly defining `home` and `workDir`. This is specifically for newt's refactor (so it's impossible for me to miss one call), and maybe doesn't make sense for `figtree`.

Reply:
Okay, I have tagged the current version at 1.0.0, so we can break api compat now and users can pin so the tag.

While we are breaking the api, lets modify the `NewFigTree` to also pass in the EnvPrefix param, so:
```
func NewFigTree(home, workDir, envPrefix string) *FigTree {
  return &FigTree{
    Home:      home,
    WorkDir:   workDir,
    EnvPrefix: envPrefix,
  }
}
```